programs/vote/src/vote_state/mod.rs

```diff
@@ -916,13 +962,49 @@ impl VoteState {
 
         self.check_slots_are_valid(vote_slots, &vote.hash, slot_hashes)?;
 
+        // Only pass voted on slot into process_next_vote_slot if the timely vote credits feature is enabled,
+        // otherwise None is passed in and that implementation will use 'normal' vote credits
+        let voted_in_slot = feature_set.and_then(|f| {
+            if f.is_active(&feature_set::timely_vote_credits::id()) {
+                Some(voted_in_slot)
+            } else {
+                None
+            }
+        });
+
         vote_slots
             .iter()
-            .for_each(|s| self.process_next_vote_slot(*s, epoch));
+            .for_each(|s| self.process_next_vote_slot(*s, epoch, voted_in_slot));
         Ok(())
     }
 
-    pub fn process_next_vote_slot(&mut self, next_vote_slot: Slot, epoch: Epoch) {
+    fn compute_vote_slot_credits(voted_in_slot: Slot, voted_for_slot: Slot) -> u64 {
+        // Should be impossible for next_vote_slot to be >= voted_in_slot (how can you vote for a slot
+        // that isn't even complete yet?) but just in case there are bugs, this check will prevent the
+        // possibility of someone exploiting that for extra credits.
```

Comment:
lets at least log something if that happens. perhaps even defaulting to 1 credit in that case.